src/main/groovy/org/agileworks/elasticsearch/river/csv/CSVRiver.java

```diff
@@ -37,7 +40,16 @@ public CSVRiver(RiverName riverName, RiverSettings settings, Client client) {
 
         config = new Configuration(settings, riverName.name());
 
-        listener = new DelegatingFileProcessorListener(this, new BashFileProcessorListener(logger, config));
+        ProcessRunnerFactory processRunnerFactory;
+
+        if (System.getProperty("os.name").toLowerCase().contains("windows")) {
```

Comment:
I did not say it clearly. I wanted to move this decision to the factory. So, there should be only one which will decide which runner will be used. Something like:

```
processRunnerFactory.getInstance().getRunner()
```